internal/applicationsnapshot/report.go

```diff
@@ -31,8 +39,9 @@ type Component struct {
 	Success    bool            `json:"success"`
 }
 
-type report struct {
-	Success    bool        `json:"success"`
+type Report struct {
```

Comment:
any reason to make this public?

Reply:
The `NewReport` function now returns an instance of Report instead of `[]byte`. I think technically this could be private, but it does seem odd to return a private type from a public function. I can change it back if you feel strongly about this.

Reply:
I guess it depends, if access (from outside of the package) to `Report` should always come from `NewReport` I would make it private. Otherwise, it's not a big deal.

Reply:
It's true. Maybe it's a controversial opinion, but I don't think a public method/function should ever have a private return type in its signature. It can make the callee code quite odd (for instance, callee can't declare a variable of that type).

If we want to hide the implementation, then I think the proper solution is to create a public interface which is used as the return type in the method/function signature. Then we can return a private type that conforms to that interface. I think this is overkill here. After all, this is all part of internal packages.